packages/firestore/test/integration/api/database.test.ts

```diff
@@ -207,6 +208,35 @@ apiDescribe('Database', persistence => {
     });
   });
 
+  it("can't overspecify field mask using set", () => {
```

Comment:
Please include a test for `[]` and for overlapping field paths, i.e. `['foo', 'foo.bar']`.

Reply:
I originally only ported the two tests that were missing from the original Android implementation. I have added more tests though, and while looking at the code noticed that the implementation didn't properly deal with field transforms. I am now also applying the mask to the transforms array. Let me know what you think and I will port it to Android.